### PR TITLE
[New Term Entry] Adding date_trunc() to SQL date terms

### PR DESCRIPTION
**Pull request by mbcrosier**
### Description

Adds a new term entry for date_trunc() - a useful PostgreSQL method - to the SQL dates section.

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above). - N/A - no associated issue

### CODE REVIEW

**Comment by SSwiniarski** [comment]
@KTom101, @yangc95, @caupolicandiaz: How do we feel about adding language elements that are specific to one flavor of SQL? I'm inclined against it.

File: content/sql/concepts/dates/terms/date-trunc/date-trunc.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  - 'learn-sql'
+  - 'paths/analyze-data-with-sql'
+  - 'paths/design-databases-with-postgresql'
+---
+
+The **`DATE_TRUNC()`** function in PostgreSQL is used to truncate a date, time, or timestamp to a specified interval, such as the day, week, or month.
+
+The `DATE_TRUNC()` function is particularly useful for time series analysis, where you want to understand how a value changes over time. Practical examples would include analyzing company's quarterly sales or determining the average hourly temperature.
+
+While the `DATE_TRUNC()` syntax is specific to PostgreSQL, SQL Server's `DATETRUNC()` function and MySQL's `EXTRACT()` function offer similar functionality.
```
Comment (mbcrosier) [inline content/sql/concepts/dates/terms/date-trunc/date-trunc.md:21]: Hi @SSwiniarski - for what it's worth, it's also available in SQL Server as datetrunc(), so could extend the entry to include that variation similar how [date_add](https://www.codecademy.com/resources/docs/sql/dates/dateadd) shows SQL Server + My SQL syntax. But also totally fine if you would rather not include it - just let me know.

**Comment by yangc95** [comment]
> How do we feel about adding language elements that are specific to one flavor of SQL? I'm inclined against it.

Only if it's specified as a note or something similar, though I can see it getting confusing. How would we break up the different flavors otherwise?

File: content/sql/concepts/dates/terms/date-trunc/date-trunc.md (hunk 1 above)
Comment (yangc95) [reply content/sql/concepts/dates/terms/date-trunc/date-trunc.md:21]: @mbcrosier Can we make a note at the top, right below the intro, with the relevant info on where it's available? cc: @SSwiniarski

Comment (mbcrosier) [reply content/sql/concepts/dates/terms/date-trunc/date-trunc.md]: hi @yangc95 - Sorry for the delay. Just moved the language details to the top and added a SQL Server example

**Review by yangc95 — APPROVED** [review]
Passing this onto @SSwiniarski for a final review

@mbcrosier I like the way you decided to break up the different flavors

**Review by SSwiniarski — APPROVED** [review]
@mbcrosier Looks good.

**Pull request merged**